tests/generate_openl3_regression_data.py

```diff
@@ -0,0 +1,414 @@
+    verbose=verbose)
+
+# compute mel256/music regression audio (6144) and image (8192) embeddings
+suffix=None
+input_repr='mel256'
+content_type='music'
+image_embedding_size=8192
+audio_embedding_size=6144
+center=True
+hop_size=0.1
+verbose=False
+
+run('video',
+    bento_path,
+    output_dir=output_dir,
+    suffix=suffix,
+    input_repr=input_repr,
+    content_type=content_type,
+    image_embedding_size=image_embedding_size,
+    audio_embedding_size=audio_embedding_size,
+    audio_center=center,
+    audio_hop_size=hop_size,    
+    verbose=verbose)
+
+# compute linear/env regression audio (512) and image (512) embeddings
+suffix='linear'
+input_repr='linear'
+content_type='env'
+image_embedding_size=512
+audio_embedding_size=512
+center=False
+hop_size=0.5
+verbose=False
+
+run('video',
+    bento_path,
+    output_dir=output_dir,
+    suffix=suffix,
+    input_repr=input_repr,
+    content_type=content_type,
+    image_embedding_size=image_embedding_size,
+    audio_embedding_size=audio_embedding_size,
+    audio_center=center,
+    audio_hop_size=hop_size,    
+    verbose=verbose)
+
+
+## Compare to previous regression data
+
+# Get separate audio and image embedding paths
+reg_audio_emb_path = os.path.join(test_path, 'data', 'regression', 'chirp_44k.npz')
+reg_audio_emb_linear_path = os.path.join(test_path, 'data', 'regression', 'chirp_44k_linear.npz')
+new_audio_emb_path = os.path.join(output_dir, 'chirp_44k.npz')
+new_audio_emb_linear_path = os.path.join(output_dir, 'chirp_44k_linear.npz')
+reg_image_emb_path = os.path.join(test_path, 'data', 'regression', 'daisy.npz')
+reg_image_emb_linear_path = os.path.join(test_path, 'data', 'regression', 'daisy_linear.npz')
+new_image_emb_path = os.path.join(output_dir, 'daisy.npz')
+new_image_emb_linear_path = os.path.join(output_dir, 'daisy_linear.npz')
+
+# Load separate audio and image embeddings
+reg_audio_emb = np.load(reg_audio_emb_path)
+reg_audio_emb_linear = np.load(reg_audio_emb_linear_path)
+new_audio_emb = np.load(new_audio_emb_path)
+new_audio_emb_linear = np.load(new_audio_emb_linear_path)
+reg_image_emb = np.load(reg_image_emb_path)
+reg_image_emb_linear = np.load(reg_image_emb_linear_path)
+new_image_emb = np.load(new_image_emb_path)
+new_image_emb_linear = np.load(new_image_emb_linear_path)
+
+assert np.allclose(reg_audio_emb['timestamps'], new_audio_emb['timestamps'])
+assert np.allclose(reg_audio_emb['embedding'], new_audio_emb['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_image_emb['embedding'], new_image_emb['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_audio_emb_linear['timestamps'], new_audio_emb_linear['timestamps'])
+assert np.allclose(reg_audio_emb_linear['embedding'], new_audio_emb_linear['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_image_emb_linear['embedding'], new_image_emb_linear['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
```

Comment:
Are regression data generation and regression tests combined into a single file here? Ideally we should have separate files for generating the original regression data (ideally never have to re-run these files), and then test files just for running the regression tests.

Reply:
They aren't, it's just a sanity check to indicate if the output has changed when re-generating the regression data.